internal/vfs/vfstest/vfstest.go

```diff
@@ -168,6 +168,29 @@ func (m *mapFS) open(p canonicalPath) (fs.File, error) {
 	return m.m.Open(string(p))
 }
 
+func (m *mapFS) remove(path string) error {
+	canonical := m.getCanonicalPath(path)
+	file, _, err := m.getFollowingSymlinks(canonical)
```

Comment:
This shouldn't run `getFollowingSymlinks`; a recursive remove will stop at any symlinks.

What might be reasonable here is to:

- Get the canonical path
- If it exists, delete it from both `m.m` and `m.symlinks`.
- If it's a dir, loop over `m.m`, and for any string that starts with `canonical+"/"`, delete it from `m.m` and `m.symlinks`.